sql-plugin/src/main/scala/org/apache/spark/sql/rapids/complexTypeCreator.scala

```diff
@@ -82,7 +82,9 @@ case class GpuCreateNamedStruct(children: Seq[Expression]) extends GpuExpression
   }.toList.unzip
 
   private lazy val names = nameExprs.map {
-    case g: GpuExpression => g.columnarEval(null)
+    case gl: GpuLiteral => gl.value
+    case ge: GpuExpression =>
+      throw new IllegalStateException(s"Unexpected GPU expression $ge")
```

Comment:
I have tried to make it so the Literal is wrapped in an Alias, which should totally be valid, but I have not been able to make it happen with the public spark APIs. If we are going to throw an exception on something we don't expect, then we should be absolutely positive that we got what we expected, and all of the default literal checks being use allow for an alias. So either still do the eval and then pull the name out of the `GpuScalar` that is returned. Or add in a second rule for an Alias of a literal, just to be on the safe side.

Reply:
> still do the eval and then pull the name out of the GpuScalar that is returned.

We can't do this because this is happening on the driver side, correct?  That's one of the drawbacks of the recent `GpuScalar` refactor -- it's practically impossible to call `columnarEval` without requiring a GPU to do it, so it's effectively off-limits to the driver.  That would be a simpler way to handle any Alias or similar shenanigans, but since we can't call `columnarEval` or `eval` to solve it, we'll have to pull it apart manually instead IIUC.

Reply:
I aggress with Jason, and the new commit has added GpuAlias case. BTW, I see the name check has been limited to literal, Do we really add the GpuAlias case?

``` scala
object CreateNamedStructCheck extends ExprChecks {
  val nameSig: TypeSig = TypeSig.lit(TypeEnum.STRING)
```

Reply:
> it's practically impossible to call columnarEval without requiring a GPU to do it, so it's effectively off-limits to the driver.

If a `GpuScalar ` is created from a Scala value (`GpuLiteral` will create such a `GpuScalar ` in its `columnarEval`) and only the `getValue` will be called to get the internal Scala value, then it is safe to use on the driver side without GPU. Becasue the GpuScalar for this case is totally a wrapper of a Scala value, nothing to do with GPU. 

But I am a little confused. Could we use the `columnarEval` on driver side with an invalid input for a generic GPU expression before the GpuScalar refactor?  I think it is still not always safe because the processing inside the `columnarEval` may require a GPU. It may be safe if we know the instance of the GPU expression. And if we know the instance of a GPU expression, I think we can use its direct methods or members instead.